sql: make pg_catalog OIDs stable and refer to the descriptor IDs directly

hueypark opened the pull request:
Feature for #32940

Can handle in future #32963, #32964

Databases and tables OID use descriptor id directly. It's fine because we create the id with sql.GenerateUniqueDescID.

Virtual tables use static negative ID. Do not use the same math.MaxUint32 anymore.

Release note(sql change): Databases and tables pg_catalog OID values are changed.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cockroachdb/cockroach?pullRequest=33420) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/cockroachdb/cockroach?pullRequest=33420) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: hueypark<br/>:white_check_mark: petermattis<br/>:x: craig[bot]<br/><hr/>**craig[bot]** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cockroachdb/cockroach?pullRequest=33420) it.</sub>

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/33420)
<!-- Reviewable:end -->

knz reviewed (commented):
Thank you for this PR. The change to give different IDs to each virtual table is very good.

I still want to investigate further how to structure the IDs so that they are not all over the place (and so that they can become stable over time). I will come back to you.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33420#-:-LVJDlO3AvFXwA0xpP82:bt2ygt8)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

hueypark committed:
sql: make pg_catalog OIDs stable and refer to the descriptor IDs directly

Databases and tables OID use descriptor id directly. It's fine because we create the id with sql.GenerateUniqueDescID.

Virtual tables use static negative ID. Do not use the same math.MaxUint32 anymore.

Release note(sql change): Databases and tables pg_catalog OID values are changed.

hueypark reviewed (commented):
Thank you for your review.

Please note that the method of assigning ID has been improved.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33420#-:-LVKKBtsBoIs_QkTTFsE:b-cpj4j6)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->
